src/app/utils/field-validation/index.ts

```diff
@@ -0,0 +1,169 @@
+
+const fieldValidatorFactory = new FieldValidatorFactory()
+
+/**
+ * Compares the response field type to the form field type
+ * @param formField The form field to compare the response to
+ * @param response The submitted response
+ */
+const isFieldTypeValid = (
+  formField: IField,
+  response: ProcessedFieldResponse,
+): Either<string, undefined> => {
+  return response.fieldType !== formField.fieldType
+    ? left(`Response fieldType (${response.fieldType}) did not match`)
+    : right(undefined)
+}
+
+/**
+ * Determines whether a response requires validation. A required field
+ * may not require an answer if it is not visible due to logic. However,
+ * if an answer is presented, it should be validated.
+ * @param formField The form field to compare the response to
+ * @param response The submitted response
+ */
+const singleAnswerRequiresValidation = (
+  formField: IField,
+  response: ProcessedSingleAnswerResponse,
+) => (formField.required && response.isVisible) || response.answer.trim() !== ''
+
+/**
+ * Generic logging function for invalid fields.
+ * Incomplete for table fields as the columnType is not logged.
+ * @param {String} formId id of form, for logging
+ * @param {Object} formField A form field from the database
+ * @param {string} message Message to log
+ * @throws {Error}
+ */
+const logInvalidAnswer = (
+  formId: string,
+  formField: IField,
+  message: string,
+) => {
+  formField.fieldType
+  logger.error({
+    message: `Invalid answer: ${message}`,
+    meta: {
+      action: 'InvalidAnswer',
+      formId,
+      fieldId: String(formField._id),
+      fieldType: formField.fieldType,
+    },
+  })
+}
+
+/**
+ * Constructs a validation function from a form field. Only meant to
+ * validate single answer responses.
+ * @param formField A form field from a form object
+ */
+const constructSingleAnswerValidator = (
+  formField: IField,
+): ResponseValidator<ProcessedSingleAnswerResponse> => {
+  if (isSectionField(formField)) {
+    return constructSectionValidator()
+  } else if (isShortTextField(formField) || isLongTextField(formField)) {
+    return constructTextValidator(formField)
+  }
+  return () => left('Unsupported field type')
+}
+
+/**
+ * Single exported function that abstracts away the complexities
+ * of field validation.
+ * @param {String} formId id of form, for logging
+ * @param {Object} formField A form field from the database
+ * @param {Object} response A client-side response that is to be untrusted
+ * @throws {Error}
+ */
+export default function validateField(
+  formId: string,
+  formField: IField,
+  response: FieldResponse,
+): void {
+  if (FIELDS_TO_REJECT.includes(response.fieldType)) {
+    throw new Error(`Rejected field type "${response.fieldType}"`)
+  }
+
+  // Validate that the form field type matches the response type
+  const fieldTypeEither = isFieldTypeValid(formField, response)
+
+  if (isLeft(fieldTypeEither)) {
+    throw new Error('Invalid field type submitted')
```

Comment:
@karrui we should definitely refactor this to throw a new `FieldValidationError` here in future, but currently the refactoring strategy aims to strictly maintain the return signature of `validateField` including the error types thrown. we can do this after all the other validators have been migrated.